src/glean.js

```diff
@@ -18,15 +18,13 @@ class Glean {
     if (browser) {
       console.log("Running from a webextension");
       appId = browser.runtime.id;
-    } else {
+    } else if (typeof document !== "undefined") {
```

Comment:
Implicit checks create a `TypeError` in QML envs. You'll notice I changed all the `if(document` and `document && ` kind of stuff to the `typeof` check.